src/internal/connector/graph/service.go

```diff
@@ -309,7 +307,7 @@ func (handler *LoggingMiddleware) Intercept(
 	msg := fmt.Sprintf("graph api error: %s", resp.Status)
 
 	// special case for supportability: log all throttling cases.
-	if resp.StatusCode == http.StatusTooManyRequests {
+	if resp.StatusCode == http.StatusTooManyRequests || resp.StatusCode == http.StatusServiceUnavailable {
```

Comment:
Letting it retry for any 500 or 503. The graph sdk retry does not happen for 503 (probably because of cumulativeDelay). For now, we blindly retry for 500 and 503 as they ideally should not be an issue from our end.